containers/daap-query-data/src/var/task/query_data.py

```diff
@@ -0,0 +1,124 @@
+import boto3
+from botocore.exceptions import ClientError
+import json
+from data_platform_api_responses import format_response_json, response_status_404
+from data_platform_logging import DataPlatformLogger
+from http import HTTPStatus
+import os
+import time
+
+
+athena_client = boto3.client('athena')
+
+def start_query_execution_and_wait(
+        athena_client,
+        database_name: str,
+        sql: str,
+        logger:DataPlatformLogger
+    ) -> str:
+        """
+        runs query for given sql and waits for completion
+        """
+        try:
+            res = athena_client.start_query_execution(
+                QueryString=sql,
+                QueryExecutionContext={"Database": database_name},
+                WorkGroup="data_product_workgroup",
+            )
+        except ClientError as e:
+            if e.response["Error"]["Code"] == "InvalidRequestException":
+                raise ValueError(e)
+            else:
+                raise ValueError(e)
+
+        query_id = res["QueryExecutionId"]
+        while response := athena_client.get_query_execution(
+            QueryExecutionId=query_id
+        ):
+            state = response["QueryExecution"]["Status"]["State"]
+            if state not in ["SUCCEEDED", "FAILED"]:
+                time.sleep(0.1)
+            else:
+                break
+
+        if not state == "SUCCEEDED":
+            logger.error(
+                "Query {}, failed with response: {}".format(
+                    sql,
+                    response["QueryExecution"]["Status"].get("StateChangeReason"),
+                )
+            )
+            raise ValueError(
+                response["QueryExecution"]["Status"].get("StateChangeReason")
+            )
+
+        return query_id
+
+
+def format_athena_results_to_table(results):
+    # Format the results into a table string
+    if not results:
+        return "No data to display"
+
+    table = ''
+
+    # Data rows
+    headers= [value for header_dict in results['ResultSet']['Rows'][0]['Data'] for value in header_dict.values()]
```

Comment:
Might need a check here to exit early if there are no rows in the result set. I don't think this is being caught by the `if not results` above.